tests/tcp_stream.rs

```diff
@@ -794,17 +795,17 @@ fn hup() {
 
 #[cfg(windows)]
 fn set_linger_zero(socket: &TcpStream) {
-    use socket2::Socket;
+    use mio::net::TcpSocket;
     use std::os::windows::io::{AsRawSocket, FromRawSocket};
 
-    let s = unsafe { Socket::from_raw_socket(socket.as_raw_socket()) };
+    let s = unsafe { TcpSocket::from_raw_socket(socket.as_raw_socket()) };
```

Comment:
Our TCP socket closes on drop, we probably need to `mem::forget` it after setting linger.

Reply:
Done